lib/Locale/Country/Extra.pm

```diff
@@ -45,22 +47,33 @@ sub idd_from_code {
     return $self->_idd_codes->{$code};
 }
 
-sub code_from_phone {
-    my ( $self, $number ) = @_;
+sub get_valid_phone {
+    my ($self, $number) = @_;
+
+    return '' if $number =~ /^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/;
 
     $number =~ s/\D//g;    # Remove non-digits
     $number =~ s/^00//;    # Remove the leading '00'.
+    return $number;
+}
+
+sub code_from_phone {
+    my ( $self, $number ) = @_;
 
-    if ( $number !~
-/^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/
-      )
-    {
-        my %code_for_idd = reverse %{ $self->_idd_codes };
-        foreach my $iddcode ( sort { $b <=> $a } keys %code_for_idd ) {
-            if ( $number =~ /^$iddcode/ ) {
-                return lc $code_for_idd{$iddcode};
-            }
-        }
+    if (my $phone = $self->get_valid_phone($number)) {
+        my %codes = %{ $self->_idd_codes };
+        return lc first{$phone =~ /^$codes{$_}/ } sort keys %codes;
```

Comment:
Just spacing:

```suggestion
        return lc first { $phone =~ /^$codes{$_}/ } sort keys %codes;
```

Though it would seem that `$self->codes_from_phone` from below could be used here instead:

```perl
if (my $first = lc ${$self->codes_from_phone($number)}[0]) {
    return $first;
}
```